src/lib/ngc/compile-source-files.ts

```diff
@@ -35,27 +33,16 @@ export async function compileSourceFiles(
 
   // Don't use `ng.emit` as it doesn't output all errors.
   // https://github.com/angular/angular/issues/24024
+  const emitFlags = tsConfigOptions.declaration ? tsConfig.emitFlags : ng.EmitFlags.JS;
+
   const result = ng.performCompilation({
     rootNames: tsConfig.rootNames,
     options: tsConfigOptions,
     host: ngCompilerHost,
     emitCallback: createEmitCallback(tsConfigOptions),
-    emitFlags: tsConfig.emitFlags
+    emitFlags
   });
 
-  // XX(hack): redirect the `*.metadata.json` to the correct outDir
```

Comment:
Note: Not needed since we can do this, with `redirect-write-file-compiler-host.ts`